modules/samtools/convert/main.nf

```diff
@@ -14,18 +13,27 @@ process SAMTOOLS_BAMTOCRAM {
     path  fai
 
     output:
-    tuple val(meta), path("*.cram"), path("*.crai"), emit: cram_crai
-    path  "versions.yml"                           , emit: versions
+    tuple val(meta), path("*.cram"), path("*.crai") , emit: cram_crai, optional: true
+    tuple val(meta), path("*.bam"), path("*.bai")   , emit: bam_bai, optional:true
+    path  "versions.yml"                            , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args  ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def file_type = input.getExtension() == "bam" ? "cram" : "bam"
```

Comment:
what about `output_extension` instead of `file_type`?

Reply:
or just plain `suffix`?